### PR TITLE
Fixes #27 - Abstract queue logic and cleanup

### PR DESCRIPTION
**Pull request by austinkelleher**
In combination with: https://github.com/windbreaker-io/windbreaker/pull/7

### CODE REVIEW

**Comment by austinkelleher** [comment]
Need to update hardcoded Rabbit url in tests to pass.

File: queue/util/createConnection.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const tri = require('tri')
+
+const CONNECTION_ATTEMPT_OPTIONS = {
+  maxAttempts: 10,
+  delay: 200,
+  factor: 2,
+  jitter: true
+}
+
+module.exports = async function creatConnection (options) {
```
Comment (charlieduong94) [inline queue/util/createConnection.js]: `createConnection`

File: queue/util/createConsumer.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  if (consumerOptions) {
+    // Dispose of the orignal connection object that was passed, so we can start
+    // with a fresh connection
+    delete consumerOptions.connection
+  }
+
+  setTimeout(async () => {
+    try {
+      await createConnection(createConsumerOptions)
```
Comment (charlieduong94) [inline queue/util/createConsumer.js]: Shouldn't we be recreating the consumer at this point?

File: test/util/MockChannel.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    return Promise.resolve()
+  }
+
+  close () {
+    if (this._closed) {
+      throw new Error('Already closed')
+    }
+    this._closed = true
+  }
+}
```
Comment (charlieduong94) [inline test/util/MockChannel.js]: Thanks for moving these out. I had them moved out in my branch as well

**Comment by charlieduong94** [comment]
[Create shared AMQ producer](https://trello.com/c/rXqMgr35/27-create-shared-amq-producer)

**Pull request merged**